vitrina/comments/views.py

```diff
@@ -74,36 +76,80 @@ def post(self, request, content_type_id, object_id):
 
             elif status := form.cleaned_data.get('status'):
+                user_org = request.user.organization
                 comment.type = Comment.STATUS
-                obj.status = status
-                obj.comment = comment.body
-                obj.save()
-                set_comment(type(obj).STATUS_CHANGED)
-
-                if isinstance(obj, Request) and status == Request.OPENED:
-                    request_plans = Plan.objects.filter(planrequest__request=obj)
-
-                    for plan in request_plans:
-                        if (
-                            plan.planrequest_set.filter(
-                                request__status=Request.OPENED
-                            ).count() == plan.planrequest_set.count() and
-                            plan.plandataset_set.annotate(
-                                has_distributions=Exists(DatasetDistribution.objects.filter(
-                                    dataset_id=OuterRef('dataset_id'),
-                                ))
-                            ).count() == plan.plandataset_set.count()
-                        ):
-                            plan.is_closed = True
-                            plan.save()
-                if obj is not None:
-                    if obj.status == 'REJECTED':
-                        email_data = prepare_email_by_identifier('application-use-case-rejected', email_content,
-                                                                 'Gautas naujas pasiūlymas',
-                                                                 [obj.title, obj.description])
-                        if obj.user is not None:
-                            if obj.user.email is not None:
-                                send_email_with_logging(email_data, [obj.user.email])
+                if isinstance(obj, Request):
+                    user_org = request.user.organization
+                    request_assignment = RequestAssignment.objects.filter(
+                        organization=user_org,
+                        request=obj
+                    ).first()
+                    obj.status = status
+                    obj.comment = comment.body
+                    obj.save()
+                    set_comment(type(obj).STATUS_CHANGED)
+                    if not request_assignment:
+                        if user_org and status==Request.APPROVED:
+                            request_assignment = RequestAssignment(
+                                request=obj,
+                                organization=user_org,
+                                status=Request.APPROVED
+                            )
+                            request_assignment.save()
+                            messages.info(request, _("Jūsų organizaciją įtraukta į poreikių organizacijų sąrašą"))
+                        else:
+                            messages.error(request, _("Jūsų organizaciją nėra įtraukta į poreikių organizacijų sąrašą"))
```

Comment:
Keista, kad pirma sukuriame komentarą, o po to tikriname ar nėra klaidų. Ar nereikėtų pirma atlikti visus patikrinimus, o tik po to keisti duomenis?

Jei organizacija nėra įtraukta, bet nurodo, kad poreikis yra patvirtintas, tada tokia organizacija turėtu būti įtraukta prie poreikio. Nereikia rodyti klaidos.